File: frontend/lib/api/tags.ts
import { api } from "./client";
import type { StockRead } from "./stocks";

export type TagCategory = "theme" | "industry_chain" | "attribute";

export interface TagRead {
  id: number;
  name: string;
  category: TagCategory;
  description?: string | null;
}

export interface StockTagRead extends TagRead {
  source: "ai" | "manual";
  confidence?: number | null;
}

export const tagsApi = {
  listAll: () => api.get<TagRead[]>("/tags"),
  listForStock: (code: string) =>
    api.get<StockTagRead[]>(`/stocks/${encodeURIComponent(code)}/tags`),
  attach: (code: string, name: string, category: TagCategory = "theme") =>
    api.post<TagRead>(`/stocks/${encodeURIComponent(code)}/tags`, {
      name,
      category,
    }),
  detach: (code: string, tagId: number) =>
    api.delete(`/stocks/${encodeURIComponent(code)}/tags/${tagId}`),
  refresh: (code: string) =>
    api.post<{ message: string }>(
      `/stocks/${encodeURIComponent(code)}/tags/refresh`,
      {},
    ),
  stocksByTag: (tagId: number) =>
    api.get<StockRead[]>(`/tags/${tagId}/stocks`),
  deleteGlobal: (tagId: number) => api.delete(`/tags/${tagId}`),
};

// 颜色映射（按 category）
export const TAG_CATEGORY_COLOR: Record<TagCategory, string> = {
  theme: "blue",
  industry_chain: "yellow",
  attribute: "green",
};

export const TAG_CATEGORY_LABEL: Record<TagCategory, string> = {
  theme: "主题",
  industry_chain: "产业链",
  attribute: "属性",
};

File: backend/app/schemas/tags.py
from typing import Literal

from pydantic import BaseModel, Field

TagCategory = Literal["theme", "industry_chain", "attribute"]


class TagRead(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    name: str
    category: TagCategory
    description: str | None = None


class StockTagRead(TagRead):
    """带挂载元信息的标签视图（详情页用）"""
    source: str = "manual"           # ai / manual
    confidence: float | None = None


class StockTagAttach(BaseModel):
    """挂载请求体：name 必填；category 在创建新标签时使用"""
    name: str = Field(..., min_length=1, max_length=64)
    category: TagCategory = "theme"
    description: str | None = None
